message K7_AppInfo
{
	required int32 id = 1;
	optional string owner = 2;
}

message K7_Login
{
	required string login = 1;
	required string password = 2;
}

message K7_UserInfo
{
	required int32 id = 1;
	required string first_name = 2;
	required string last_name = 3;
	optional K7_Login login = 4;
}

message K7_SignIn
{
	required string app_name = 1;
	required K7_UserInfo user = 2;
}

message K7_BalanceAnswer
{
	required K7_AppInfo app = 1;
	required int32 result = 2;
}

service K7TalkServer
{
  rpc login(K7_Login) returns(K7_UserInfo);
  rpc signin_app(K7_SignIn) returns(K7_AppInfo);
  rpc req_increase(K7_AppInfo) returns(K7_BalanceAnswer);
  rpc req_decrease(K7_AppInfo) returns(K7_BalanceAnswer);
}
